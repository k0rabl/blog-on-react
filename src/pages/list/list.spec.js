import React from "react"
import ConnectedList, { List } from './index'
import { mount } from 'enzyme';

import {
  BrowserRouter,
  Router
} from "react-router-dom"

import configureStore from 'redux-mock-store'
import { Provider } from "react-redux";
// import { handleRead, handleEditElement, handleAddElement } from '../../features/Article/ArticleSlice'

import Articles from "../../fixtures/Articles";
import {createMemoryHistory} from 'history'


describe('<List />', () => {
  const initialState = { 
    search: {        
      articles: Articles,
    },
    edit: {        
      editMode: true,
    },
    pagination: {
      active: 1
    }
  };

  const initialStateWtihOne = { 
    search: {        
      articles: [Articles[0]],
    },
    edit: {        
      editMode: true,
    },
    pagination: {
      active: 1
    }
  };
  const mockStore = configureStore();
  let store, component, history;
  beforeEach(() => {
    store = mockStore(initialState)
    history = createMemoryHistory()

    component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedList/>
        </BrowserRouter>
      </Provider>
    )
  })

  
  it('+++ render the component', () => {
    expect(component.find(ConnectedList).length).toEqual(1)
  })

  it('+++ render the component with 1 article', () => {
    store = mockStore(initialStateWtihOne)

    const componentSearch = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedList />
        </BrowserRouter>
      </Provider>
    )

    expect(componentSearch.find(ConnectedList).length).toEqual(1)
    expect(componentSearch).toMatchSnapshot()
  })
  
  it('+++ check text search', () => {
    history.push('/search?string=how');
    
    const componentSearch = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedList />
        </Router>
      </Provider>
    )
    
    expect(componentSearch.find(ConnectedList).length).toEqual(1)
    expect(componentSearch).toMatchSnapshot()
  })

  it('+++ check calendar search', () => {
    history.push('/search?date=2021-12-10');
    
    component = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedList />
        </Router>
      </Provider>
    )
    
    expect(component.find(ConnectedList).length).toEqual(1)
    expect(component).toMatchSnapshot()
  })
})