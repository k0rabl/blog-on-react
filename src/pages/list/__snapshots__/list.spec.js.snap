// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> +++ render the component with 1 article 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(List))>
        <Connect(List)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <List
            active={1}
            articles={
              Array [
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 1,
                  "image": "",
                  "isRead": true,
                  "name": "How to use hooks in React?",
                  "preview": "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
                },
              ]
            }
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            setActive={[Function]}
          >
            <div
              className="list"
            >
              <Connect(Edit)>
                <Edit
                  editMode={true}
                  handleMode={[Function]}
                >
                  <div
                    className="editMode__container editMode__container-edit"
                  >
                    <Link
                      className="waves-effect waves-light redC-bg lighten-3 btn btn-add button"
                      to="/add"
                    >
                      <LinkAnchor
                        className="waves-effect waves-light redC-bg lighten-3 btn btn-add button"
                        href="/add"
                        navigate={[Function]}
                      >
                        <a
                          className="waves-effect waves-light redC-bg lighten-3 btn btn-add button"
                          href="/add"
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons left"
                          >
                            add
                          </i>
                          Add article
                        </a>
                      </LinkAnchor>
                    </Link>
                    <button
                      className="btn-close button"
                      onClick={[Function]}
                    >
                      <i
                        className="material-icons left"
                      >
                        close
                      </i>
                      Exit
                    </button>
                  </div>
                </Edit>
              </Connect(Edit)>
              <Connect(Article)
                date="2021-05-20"
                desc="Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum."
                id={1}
                image=""
                isRead={true}
                key="1"
                name="How to use hooks in React?"
                preview="Lorem ipsum dolor sit amet consectetur adipisicing elit."
              >
                <Article
                  date="2021-05-20"
                  desc="Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum."
                  editMode={true}
                  handleDeleteElement={[Function]}
                  id={1}
                  image=""
                  isRead={true}
                  name="How to use hooks in React?"
                  preview="Lorem ipsum dolor sit amet consectetur adipisicing elit."
                >
                  <div
                    className="container article"
                  >
                    <div
                      className="top"
                    >
                      <div
                        className="top-left"
                      >
                        <Link
                          className="link articleLink"
                          to="/1"
                        >
                          <LinkAnchor
                            className="link articleLink"
                            href="/1"
                            navigate={[Function]}
                          >
                            <a
                              className="link articleLink"
                              href="/1"
                              onClick={[Function]}
                            >
                              <div
                                className="title read edit"
                              >
                                How to use hooks in React?
                              </div>
                            </a>
                          </LinkAnchor>
                        </Link>
                        <div
                          className="buttons"
                        >
                          <button
                            className="button btn-delete"
                            onClick={[Function]}
                          >
                            <i
                              className="material-icons center"
                            >
                              delete
                            </i>
                          </button>
                        </div>
                      </div>
                      <div
                        className="date"
                      >
                        2021-05-20
                      </div>
                    </div>
                    <div
                      className="bottom"
                    >
                      <div
                        className="leftImage"
                      >
                        <span>
                          No image
                        </span>
                      </div>
                      <div
                        className="rightDescr"
                      >
                        <div
                          className="descr"
                        >
                          Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.
                        </div>
                      </div>
                    </div>
                  </div>
                </Article>
              </Connect(Article)>
            </div>
          </List>
        </Connect(List)>
      </withRouter(Connect(List))>
    </Router>
  </BrowserRouter>
</Provider>
`;
